leap_c/rl/sac_fop.py

```diff
@@ -224,42 +182,37 @@ def init_policy_state(self) -> Any:
     def train_loop(self) -> Iterator[int]:
         is_terminated = is_truncated = True
         episode_return = episode_length = np.inf
+        episode_act_stats = defaultdict(list)
         policy_state = self.init_policy_state()
-        mpc_stats_aggregated_rollout = {}
+        obs = None
 
         while True:
             if is_terminated or is_truncated:
                 obs, _ = self.train_env.reset()
-                if (
-                    episode_length < np.inf
-                ):  # TODO: Add rollout-logging for the non-episodic case
-                    stats = {}
-                    for k, v in mpc_stats_aggregated_rollout.items():
-                        if not ("max" in k or "min" in k):
-                            stats["avg_" + k] = v / episode_length
-                        else:
-                            stats[k] = v
-
-                    stats["episode_return"] = episode_return
-                    stats["episode_length"] = episode_length
-                    self.report_stats("train_rollout", stats, self.state.step)
+                if episode_length < np.inf:
+                    mpc_episode_stats = {k: float(np.mean(v)) for k, v in episode_act_stats.items()}
+                    episode_stats = {
+                        "episode_return": episode_return,
+                        "episode_length": episode_length,
+                    }
+                    self.report_stats("train", episode_stats, self.state.step)
+                    self.report_stats("train_policy_rollout", mpc_episode_stats, self.state.step)
                 policy_state = self.init_policy_state()
-                mpc_stats_aggregated_rollout = {}
                 is_terminated = is_truncated = False
                 episode_return = episode_length = 0
-            action, policy_state_prime, param, status, mpc_stats = self.act(
-                obs, state=policy_state
-            )
+                episode_act_stats = defaultdict(list)
+
+            action, policy_state_prime, act_stats = self.act(obs, state=policy_state)
```

Comment:
I would prefer this looked like in sac_zop, it is more consistent that way and information is more accessible, in case one needs to do something with it.